src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,243 +32,218 @@
-    output_dir: Path
-        path of dir to which the zip content should be written
-    use_cache: bool
-        if True, don't download if output_dir already exists
-
-    Returns
-    -------
-    country_data : str
-        if data found return the output_dir, else return None
+    country_config : CountryConfig
+        Country configuration
     """
-    url_country_date = _BASE_URL_COUNTRY.format(
-        iso2=iso2.upper(), YYYY=date_pub.year, MM=date_pub.month
-    )
-
-    # filenames have upper iso2, so use that for dirs as well
-    output_dir_country = (
-        output_dir / f"{iso2.upper()}{date_pub.strftime('%Y%m')}"
-    )
-    zip_filename = f"{output_dir_country.name}.zip"
-    if not output_dir_country.exists() or use_cache is False:
-        country_data = _download_zip(
-            url=url_country_date,
-            zip_filename=zip_filename,
-            output_dir=output_dir_country,
+
+    def __init__(
+        self,
+        country_config,
+    ):
+        super().__init__(
+            country_config=country_config,
+            module_base_dir=_MODULE_BASENAME,
+            is_public=True,
         )
-    else:
-        country_data = output_dir_country
-    return country_data
-
-
-def _download_fewsnet_region(
-    date_pub: date,
-    region_name: str,
-    region_code: str,
-    output_dir: Path,
-    use_cache: bool,
-) -> Optional[Path]:
-    """
-    Download fewsnet data that covers `region_name`.
 
-    Parameters
-    ----------
-    date_pub : date
-        date for which the data should be downloaded
-        only the year and month part are used
-        this commonly refers to the month of the Current Situation period
-    region_name : str
-        name of the region to which the `iso2` belongs,
-        e.g. "east-africa"
-    region_code : str
-        code that refers to the `region_name,
-        e.g. "EA"
-    output_dir : Path
-        path to dir to which the data should be written
-    use_cache : bool
-        if True, don't download if output_dir already exists
-
-    Returns
-    -------
-    region_data : str
-        If region data exists, return the saved dir else return None
-    """
-    url_region_date = _BASE_URL_REGION.format(
-        region_code=region_code.upper(),
-        region_name=region_name,
-        YYYY=date_pub.year,
-        MM=date_pub.month,
-    )
-
-    output_dir_region = (
-        output_dir / f"{region_code.upper()}{date_pub.strftime('%Y%m')}"
-    )
-    zip_filename_region = f"{output_dir_region.name}.zip"
-    if not output_dir_region.exists() or use_cache is False:
-        region_data = _download_zip(
-            url=url_region_date,
-            zip_filename=zip_filename_region,
-            output_dir=output_dir_region,
+        # FN data can be regional, and thus we save the raw data
+        # in "glb" instead of in the iso3 folder, even though part of it is at
+        # country level
+        self._raw_base_dir = self._get_base_dir(
```

Comment:
Yeah it might be a good idea (though not applicable here yet as we didn't implement `process`).
I am just wondering if we ever want to save processed global data, what you think?

Reply:
Anyway especially in view of other users it might be better to allow for that

Reply:
> I am just wondering if we ever want to save processed global data, what you think?

Hm yeah, if you can't think of any examples then I certainly can't. However, could be good to have for flexibility when subclassing DataSource in the pilot repos.